core/pool_vector.h

```diff
@@ -442,14 +442,51 @@ class PoolVector {
 	}
 
 	String join(String delimiter) {
-		String rs = "";
```

Comment:
Instead of copying the implementation of the `StringBuilder`, can't you *use* it instead? The code should be a lot simpler then.

You could basically keep the old code but swap this out with `StringBuilder` instead of `String` here.

Reply:
That was my first idea, too. But esp since we have a delimiter here, I decided to go with a more specialized approach.

Reply:
I would say it's worth it to keep this implementation in a single place.

Have you tried using `StringBuilder` and comparing it to your solution? If the performance characteristics are decent enough I would say that the easier code should be used.

Reply:
No, I did not. I think the implementation is still straight forward. StringBuilder covers a lot of different cases which puts limits on its implementation which are not required, here.

Reply:
The only different case is that it handles constant C strings as well, which shouldn't cost much if you only add `String`s, I personally think the implementation should not be copied for this case.

`StringBuilder` was added exactly for cases like this, I would like to ask you to try out an implementation with it and run the same benchmarks.

Reply:
I did a test using the StringBuilder and the results are even worse...

I will do some testing by using the StringBuilder for the join() and try fixing the StringBuilder implementation, 'instead'.